src/models/task.model.ts

```diff
@@ -0,0 +1,76 @@
+import { Schema } from 'mongoose';
+import { TaskPriorityMapping } from '../repositories/task/create';
+import { generateId } from '../utils/utils';
+
+interface NotificationSubscriptionSchema {
+  reminder: boolean;
+  taskStarted: boolean;
+  taskCompleted: boolean;
+}
+
+type PriorityLevelKeyType = keyof typeof TaskPriorityMapping;
+type PriorityLevelValueType = (typeof TaskPriorityMapping)[PriorityLevelKeyType];
+
+export interface TaskModel {
+  id: string;
+  title: string;
+  description?: string;
+  startTime?: Date;
+  createdAt: Date;
+  updatedAt: Date;
+  deletedAt?: Date;
+  completionTime?: Date;
+  priorityLevel?: PriorityLevelValueType;
+  notificationSubscription?: NotificationSubscriptionSchema;
+  __v: number;
```

Comment:
Do you need to add this field?

Reply:
I added it here because TS raised that it doesn't exist on TaskModel when I tried to restructure it in the "create" repository. If I didn't do so, I might be forced to return it from the repository.